elastic/kibana
# [kbn-es] add required flag projectType for serverless script

## Summary

in https://github.com/elastic/kibana/pull/174284 we split serverless roles into individual files per project.

If you run `yarn es serverless --ssl` ES will be provisioned only with roles specified for elastisearch project type.
To use roles for oblt/security projects, you have to override roles file with `--resources` flag:
```
yarn es serverless --ssl --resources /Users/dmle/github/kibana/packages/kbn-es/src/serverless_resources/project_roles/security/roles.yml
```

Since it is confusing and not dev-friendly approach, this PR adds new required flag to `serverless` script: `--projectType`

Usage:
`yarn es serverless --projectType=es --ssl`
`yarn es --serverless=oblt --ssl`

roles.yml file will be picked up based on `projectType` value, you still have an option to override it using `--resources` flag

How to test:
`yarn es serverless --project-type=oblt --ssl`
`yarn start serverless=oblt --ssl`

You should be able to login with all roles defined or Observability (and other) project.

Cli docs were updated:
```
[main][~/github/kibana]$ node scripts/es --help
usage: es <command> [<args>]

Assists with running Elasticsearch for Kibana development

Available commands:

  snapshot - Downloads and run from a nightly snapshot
  source - Build and run from source
  archive - Install and run from an Elasticsearch tar
  build_snapshots - Build and collect ES snapshots
  docker - Run an Elasticsearch Docker image
  serverless - Run Serverless Elasticsearch through Docker

To start a serverless instance use the 'serverless' command with
  '--projectType' flag or use the '--serverless=<ProjectType>'
  shortcut, for example:

    es --serverless=es

Global options:

  --help
```
